### PR TITLE
simplified public tests

### PR DESCRIPTION
**Pull request by doozMen**
I think it is important to test the public interface without the need for @testable

### CODE REVIEW

File: Tests/SwiftFSMTests/FSMTests/SimplifiedPublicTests.swift [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import XCTest
+import SwiftFSM
+
+final class SimplifiedPublicTests: XCTestCase {
```
Comment (drseg) [inline Tests/SwiftFSMTests/FSMTests/SimplifiedPublicTests.swift:4]: Change file and test class name to PublicInterfaceTests, to accompany some changes to other file names that I made in preparation for this (PublicTests -> PublicFSMTests, FSMTests -> InternalFSMTests).

File: Tests/SwiftFSMTests/FSMTests/SimplifiedPublicTests.swift [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    var lockCalled = false
+    var alarmCalled = false
+    var thankyouCalled = false
+    
+    func unlock() {
+      unlockCalled = true
+    }
+    func alarm() { alarmCalled = true }
+    func thankyou() { thankyouCalled = true }
+    func lock() { lockCalled = true }
```
Comment (drseg) [inline Tests/SwiftFSMTests/FSMTests/SimplifiedPublicTests.swift:35]: Inline the unlock() body

File: Tests/SwiftFSMTests/FSMTests/SimplifiedPublicTests.swift [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    XCTAssertTrue(sut.unlockCalled)
+    XCTAssertFalse(sut.thankyouCalled)
+    try sut.fsm.handleEvent(.coin)
+    XCTAssertTrue(sut.thankyouCalled)
+    try sut.fsm.handleEvent(.coin)
+    XCTAssertFalse(sut.lockCalled)
+    try sut.fsm.handleEvent(.pass)
+    XCTAssertTrue(sut.lockCalled)
+    try sut.fsm.handleEvent(.pass)
+    XCTAssertTrue(sut.alarmCalled)
```
Comment (drseg) [inline Tests/SwiftFSMTests/FSMTests/SimplifiedPublicTests.swift:51]: Add a blank line after each handleEvent()/XCTAssert... pair

File: Tests/SwiftFSMTests/FSMTests/SimplifiedPublicTests.swift
Comment (drseg) [inline Tests/SwiftFSMTests/FSMTests/SimplifiedPublicTests.swift:1]: Update indents to four spaces to match rest of project

File: Tests/SwiftFSMTests/FSMTests/SimplifiedPublicTests.swift (hunk 1 above)
Comment (drseg) [reply Tests/SwiftFSMTests/FSMTests/SimplifiedPublicTests.swift:4]: (make sure to pull the latest changes from master first)

**Review by drseg — CHANGES_REQUESTED** [review]
Thank you very much for doing this. You are correct of course, and I would have saved myself some issues if I'd done this much earlier!

I've left a few comments as requests for changes just to make things match up with the rest of the project and then I'd be happy to put it in.

File: Tests/SwiftFSMTests/FSMTests/SimplifiedPublicTests.swift (hunk 2 above)
Comment (doozMen) [reply Tests/SwiftFSMTests/FSMTests/SimplifiedPublicTests.swift]: ok done

File: Tests/SwiftFSMTests/FSMTests/SimplifiedPublicTests.swift
Comment (doozMen) [reply Tests/SwiftFSMTests/FSMTests/SimplifiedPublicTests.swift:1]: oh yes my settings are different, keep forgetting that. Changed it!

**Review by drseg — APPROVED** [review]

**Pull request merged**